### PR TITLE
Windows Custom Cluster Automation

### PR DESCRIPTION
**Pull request by jameson-mcghee**
[QA-Task 428](https://github.com/rancher/qa-tasks/issues/428)

## Problem
Currently all testing of the provisioning of Custom clusters with Windows worker nodes has to be handled manually or with bash scripts, therefore having the ability to run automated CI/CD tests for the provisioning of these clusters to ensure they are still working properly will be much more efficient on time.

## Solution
This PR will allow for automation of Windows Custom Cluster nodes, to allow expedited regression/smoke testing.

## Automated Testing
This PR is intended to add automated tests for the provisioning of Custom Clusters with Windows worker nodes. The additions of new Windows provisioning tests (1 Node all roles Admin User + 1 Windows Worker, 1 Node all roles Standard User + 1 Windows Worker, 3 unique role nodes as Admin User + 1 Windows Worker, 3 unique role nodes as Standard User + 1 Windows Worker) will cover basic regression testing scenarios.

**Note:** Running this suite of tests for Custom Clusters (including bother with and without Windows Nodes) often takes ~90+ minutes to complete.

### CODE REVIEW

**Comment by igomez06** [comment]
@jameson-mcghee does this PR have a backport as well?

**Comment by jameson-mcghee** [comment]
@igomez06, I have not created a backport yet, because I wanted to verify that this one was going to get approval first, but we can if thats what is needed.

**Comment by igomez06** [comment]
> @igomez06, I have not created a backport yet, because I wanted to verify that this one was going to get approval first, but we can if thats what is needed.

For sure, but this is practically approved just one more review with a Jenkins run, so I think it's a good time to get a backport ready. That way both PRs can be merged at the same time.

**Comment by markusewalker** [comment]
I would agree with @igomez06 on the backport. The PR can be merged once we have the last Jenkins run, so it would be good to just cherry-pick this into the 2.6 branch and knock two birds out with one stone.

**Review by markusewalker — DISMISSED** [review]
LGTM.

File: tests/v2/validation/provisioning/nodeproviders.go
Comment (igomez06) [inline tests/v2/validation/provisioning/nodeproviders.go:61]: One little nitpick, this could be done with an else.` if () {}else{}`

**Review by igomez06 — COMMENTED** [review]
Looks great, I just had one little nitpick.

**Review by slickwarren — APPROVED** [review]
LGTM, nice work!

File: tests/v2/validation/provisioning/nodeproviders.go
Comment (jameson-mcghee) [reply tests/v2/validation/provisioning/nodeproviders.go]: Done

**Review by igomez06 — APPROVED** [review]
LGTM

**Review by markusewalker — APPROVED** [review]

**Review by slickwarren — DISMISSED** [review]
LGTM

**Review by markusewalker — APPROVED** [review]

**Pull request merged**
